bin/dev-tools/create_shipments.py

```diff
@@ -0,0 +1,332 @@
+        if self.users['user_1'] not in (self.moderator, self.carrier, self.shipper):
+            self.active_users.remove('user_1')
+
+    def _chunker(self, array):
+        return (array[i:i + self.chunk_size] for i in range(0, len(array), self.chunk_size))
+
+    def _validate_user(self, user):
+        if user not in self.users:
+            self._process_failure(f'Invalid user: {user}')
+        if user not in self.active_users:
+            self.active_users.add(user)
+        return self.users[user]
+
+    def _validate_attributes(self, attributes):
+        if self.verbose:
+            print(f'Validating attributes: {attributes}')
+        for attribute in attributes:
+            try:
+                attribute = json.loads(attribute)
+            except JSONDecodeError:
+                if self.verbose:
+                    print(f'Non json attribute recieved: {attribute}')
+            if isinstance(attribute, dict):
+                self.attributes.update(attribute)
+            else:
+                try:
+                    key, value = attribute.split("=")
+                except ValueError:
+                    self._process_failure(f'Invalid format for attribute: {attribute}, should be in format: key=value')
+                self.attributes[key] = value
+
+    def _process_failure(self, error_message):
+        print(f'Shipment Creator failed: {error_message}')
+        print(f'Response errors: {self.errors}')
+        sys.exit(2)
+
+    def _parse_request(self, url, method='post', **kwargs):
+        if method not in ['post', 'get', 'patch']:
+            self._process_failure(f'Invalid method: {method}')
+        request_method = getattr(requests, method)
+        try:
+            response = request_method(url, **kwargs)
+        except requests.exceptions.ConnectionError:
+            if self.verbose:
+                print(f'Connection error raised when connecting to {url}')
+            self._process_failure(f'Error connecting to url: {url}')
+
+        if not response.ok:
+            if self.verbose:
+                print(f'Invalid response returned from {url}')
+            self.errors += response.json()['errors']
+            return None
+        return response.json()
+
+    def _retrieve_updated_attributes(self):
+        if self.verbose:
+            print(f'Updating attributes to remove update random and sequential variables')
+        updated_attributes = deepcopy(self.attributes)
+        for key, value in updated_attributes.items():
+            if value == "##RAND##":
+                updated_attributes[key] = str(uuid4())
+            elif value == "##NUM##":
+                updated_attributes[key] = self.sequence_number
+        if self.device:
+            device_response = self._parse_request(
+                f"{self.profiles_url}/api/v1/device",
+                data={'device_type': "AXLE_GATEWAY"},
+                headers={'Authorization': 'JWT {}'.format(self.shipper['token'])})
```

Comment:
Instead of using the token directly here, it may be better to directly call `get_user_jwt` here.  That method could then be merged with the logic in `_set_user_tokens` so that tokens are generated (if not present, about to expire, or expired) or returned from the token cache for the user.  This change would remove the requirement for calling `_set_user_tokens` at various places in your code and would more directly validate the tokens existence and validity at usage time instead of at some unknown time prior.